### PR TITLE
Feature/work3

### PR DESCRIPTION
**Pull request by Uaplexer**
(not fully completed third task but please check if smthing in this exact code is wrong)

### CODE REVIEW

File: HomeWork2/main.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
+import requests
```
Comment (vloooo) [inline HomeWork2/main.py]: why task 2 is here?

File: HomeWork3/main.py [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+
+
+def set_logger(logging_level):
+    logging.basicConfig(filename='log.txt',
+                        level=logging_level.upper(),
+                        format='%(asctime)s:%(levelname)s:%(name)s:%(message)s')
+    return logging.getLogger('my_logger')
+
+
+num_of_results = 5000
```
Comment (vloooo) [inline HomeWork3/main.py]: is it a constant?

File: HomeWork3/main.py [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+
+
+def parse_args():
+    parser = argp.ArgumentParser()
+    exclusive_group = parser.add_mutually_exclusive_group()
+    exclusive_group.add_argument('--gender', type=str, help='Sort users by gender')
+    exclusive_group.add_argument('--num_of_rows', type=int, help='Numbers of total user rows')
+    parser.add_argument('--path', type=str, help='Path to destination dir')
+    parser.add_argument('--filename', type=str, default='output', help='Name of data file')
+    parser.add_argument('--log_level', type=str, help='Choose one of logging levels:INFO,WARNING')
```
Comment (vloooo) [inline HomeWork3/main.py]: should be positional

File: HomeWork3/main.py [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+from statistics import mean
+
+
+def parse_args():
+    parser = argp.ArgumentParser()
+    exclusive_group = parser.add_mutually_exclusive_group()
+    exclusive_group.add_argument('--gender', type=str, help='Sort users by gender')
+    exclusive_group.add_argument('--num_of_rows', type=int, help='Numbers of total user rows')
+    parser.add_argument('--path', type=str, help='Path to destination dir')
+    parser.add_argument('--filename', type=str, default='output', help='Name of data file')
```
Comment (vloooo) [inline HomeWork3/main.py:18]: mutually exclusive group

Comment (vloooo) [reply HomeWork3/main.py:18]: ooops

File: HomeWork3/main.py [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+from collections import Counter
+from statistics import mean
+
+
+def parse_args():
+    parser = argp.ArgumentParser()
+    exclusive_group = parser.add_mutually_exclusive_group()
+    exclusive_group.add_argument('--gender', type=str, help='Sort users by gender')
+    exclusive_group.add_argument('--num_of_rows', type=int, help='Numbers of total user rows')
+    parser.add_argument('--path', type=str, help='Path to destination dir')
```
Comment (vloooo) [inline HomeWork3/main.py]: should be required

File: HomeWork3/main.py [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+
+def set_logger(logging_level):
+    logging.basicConfig(filename='log.txt',
+                        level=logging_level.upper(),
+                        format='%(asctime)s:%(levelname)s:%(name)s:%(message)s')
+    return logging.getLogger('my_logger')
+
+
+num_of_results = 5000
+URL = f'https://randomuser.me/api/?results={num_of_results}'
```
Comment (vloooo) [inline HomeWork3/main.py]: where have u used it?

File: HomeWork3/main.py [hunk 7]
```diff
@@ -0,0 +1,239 @@
...
+        file.write((requests.get(url)).text)
+    logging.info('Successfully downloaded csv')
+
+
+def filter_csv(csv_data, gender=None, num_of_rows=None):
+    filtered_data = []
+    if gender:
+        filtered_data = list(filter(lambda x: x['gender'] == gender, csv_data))
+        logger.info('Data was filtered by gender')
+    if num_of_rows and num_of_rows <= len(csv_data):
```
Comment (vloooo) [inline HomeWork3/main.py]: elif?

File: HomeWork3/main.py [hunk 8]
```diff
@@ -0,0 +1,239 @@
...
+        logger.warning("Data wasn't filtered by any of params")
+    return filtered_data
+
+
+def current_time_implementation(location_timezone_offset):
+    if location_timezone_offset == '0:00':
+        return datetime.utcnow()
+    else:
+        hours, minutes = map(int, location_timezone_offset[1:].split(':'))
+        sign = 1 if location_timezone_offset.startswith('+') else -1
```
Comment (vloooo) [inline HomeWork3/main.py]: check if you need to operate sign

File: HomeWork3/main.py [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+        filtered_data = csv_data[:num_of_rows]
+        logger.info('Data was filtered by number of rows')
+    else:
+        logger.warning("Data wasn't filtered by any of params")
+    return filtered_data
+
+
+def current_time_implementation(location_timezone_offset):
+    if location_timezone_offset == '0:00':
+        return datetime.utcnow()
```
Comment (vloooo) [inline HomeWork3/main.py]: I would skip this part. if u have zeros, your time delta just will add nothing

File: HomeWork3/main.py [hunk 10]
```diff
@@ -0,0 +1,239 @@
...
+
+
+def add_and_change_fields_in_file(csv_data):
+    for i, elem in enumerate(csv_data, start=1):
+        elem['global_index'] = i
+        elem['current_time'] = current_time_implementation(elem['location.timezone.offset'])
+        elem['name.title'] = name_title_implementation(elem['name.title'])
+        elem['dob.date'] = datetime.strptime(elem['dob.date'][:10], '%Y-%m-%d').strftime('%m/%d/%Y')
+        elem['registered.date'] = datetime.strptime(elem['registered.date'][:19], '%Y-%m-%dT%H:%M:%S').strftime(
+            '%m-%d-%Y, %H:%M:%S')
```
Comment (vloooo) [inline HomeWork3/main.py]: do you really need slicing here?

File: HomeWork3/main.py [hunk 11]
```diff
@@ -0,0 +1,239 @@
...
+        curr_path_folder = os.getcwd() + '\\'
+        os.chdir('C:\\Program Files\\')
+        logger.info('Successfully changed directory to C:\\Program Files\\')
+        os.makedirs("Users data folder", exist_ok=True)
+        logger.info('Successfully created folder "Users data folder"')
+        os.chdir('C:\\Program Files\\Users data folder')
+        logger.info('Successfully changed folder to "C:\\Program Files\\Users data folder"')
+        move(curr_path_folder + EXACT_CSV_FILENAME, 'C:\\Program Files\\Users data folder')
+        logger.info('Successfully moved file to default directory "C:\\Program Files\\Users data folder"')
+        path_to_dir = 'C:\\Program Files\\Users data folder'
```
Comment (vloooo) [inline HomeWork3/main.py]: this  part is unreadable. at least break logic blocks of code by blank lines

File: HomeWork3/main.py [hunk 12]
```diff
@@ -0,0 +1,239 @@
...
+        elem['name.title'] = name_title_implementation(elem['name.title'])
+        elem['dob.date'] = datetime.strptime(elem['dob.date'][:10], '%Y-%m-%d').strftime('%m/%d/%Y')
+        elem['registered.date'] = datetime.strptime(elem['registered.date'][:19], '%Y-%m-%dT%H:%M:%S').strftime(
+            '%m-%d-%Y, %H:%M:%S')
+    logger.info('Successfully added and modified attributes')
+    return csv_data
+
+
+def create_and_move_file_to_destination_folder(path_to_dir=None):
+    if not path_to_dir:
```
Comment (vloooo) [inline HomeWork3/main.py]: path to dir should be required parameter in args

File: HomeWork3/main.py [hunk 13]
```diff
@@ -0,0 +1,239 @@
...
+        logger.warning('No path was entered in args!')
+        curr_path_folder = os.getcwd() + '\\'
```
Comment (vloooo) [inline HomeWork3/main.py]: you used it just once. better to add os.path.join directly in line 112.

File: HomeWork3/main.py [hunk 14]
```diff
@@ -0,0 +1,239 @@
...
+        os.chdir('C:\\Program Files\\')
+        logger.info('Successfully changed directory to C:\\Program Files\\')
+        os.makedirs("Users data folder", exist_ok=True)
```
Comment (vloooo) [inline HomeWork3/main.py]: double-quotes.
never use file and folder names with spaces. replace them with underscores

File: HomeWork3/main.py [hunk 15]
```diff
@@ -0,0 +1,239 @@
...
+        logger.info('Successfully created folder "Users data folder"')
+        os.chdir('C:\\Program Files\\Users data folder')
+        logger.info('Successfully changed folder to "C:\\Program Files\\Users data folder"')
+        move(curr_path_folder + EXACT_CSV_FILENAME, 'C:\\Program Files\\Users data folder')
```
Comment (vloooo) [inline HomeWork3/main.py]: don't use **+** concatenation. better to apply f-strings

File: HomeWork3/main.py [hunk 16]
```diff
@@ -0,0 +1,239 @@
...
+        os.chdir(path_to_dir)
+        os.makedirs("Users data folder", exist_ok=True)
+        os.chdir(path_to_dir + '\\Users data folder')
+    return path_to_dir
+
+
+def rearrange_data(csv_data):
+    new_data = {}
+    for elem in csv_data:
+        new_data.setdefault(f'{elem["dob.date"][-2]}0-th', {})
```
Comment (vloooo) [inline HomeWork3/main.py]: u used f'{elem["dob.date"][-2]}0-th' 3 times.
move to a variable and reuse

File: HomeWork3/main.py [hunk 17]
```diff
@@ -0,0 +1,239 @@
...
+        os.chdir(curr_dir + '\\' + i)
+        for elem in COUNTRIES:
+            os.makedirs(elem)
+    logger.info('Successfully created sub folders with Countries')
+    os.chdir(curr_dir)
+
+
+def create_sub_folders_decades():
+    for i in DECADES:
+        os.makedirs(i)
```
Comment (vloooo) [inline HomeWork3/main.py]: should be rewritten. let's discuss

File: HomeWork3/main.py [hunk 18]
```diff
@@ -0,0 +1,239 @@
...
+def delete_decades_under_60():
+    for decades in DECADES:
+        if int(decades[:2]) < 60 and not 00:
+            rmtree(os.getcwd() + '\\' + decades[:2] + '-th')
+            logger.info(f'Successfully deleted {decades[:2] + "-th"} decade')
+
+
+# This func is creating infinite zip
+def archive_dest_folder(path_to_folder):
+    if path_to_folder:
```
Comment (vloooo) [inline HomeWork3/main.py]: it's required

File: HomeWork3/main.py [hunk 19]
```diff
@@ -0,0 +1,239 @@
...
+filtered = filter_csv(initial_csv_data, gender=args.gender, num_of_rows=args.num_of_rows)
+# Task 5
+modified_data = add_and_change_fields_in_file(filtered)
+write_to_csv(modified_data, EXACT_CSV_FILENAME)
+# Task 6-7
+create_and_move_file_to_destination_folder(args.path)
+# Task 8
+rearranged = rearrange_data(modified_data)
+# Task 9
+DECADES = get_decades(rearranged)
```
Comment (vloooo) [inline HomeWork3/main.py]: it's cannot be constant. you explicitly received it from calculation

File: HomeWork3/main.py [hunk 20]
```diff
@@ -0,0 +1,239 @@
...
+create_sub_folders_countries()
+# Task 11
+write_users_data_to_different_files(rearranged)
+# Task 12
+delete_decades_under_60()
+
+# Task 14 (func is creating infinite zip)
+archive_dest_folder(args.path)
+
+# Task is not fully completed, i'll complete it fully in a few days!!!!
```
Comment (vloooo) [inline HomeWork3/main.py]: like

File: HomeWork3/main.py [hunk 21]
```diff
@@ -0,0 +1,239 @@
...
+            except KeyError:
+                logger.warning(f'No country {country} in {decade}')
+    logger.info('Successfully wrote users data to local files')
+    os.chdir(curr_dir)
+
+
+def delete_decades_under_60():
+    for decades in DECADES:
+        if int(decades[:2]) < 60 and not 00:
+            rmtree(os.getcwd() + '\\' + decades[:2] + '-th')
```
Comment (vloooo) [inline HomeWork3/main.py]: os.path.join
it's funny) decades contain a string like 70-th. you apply slicing to extract digit part and then add -th))))
use f-strings for concatenation

File: HomeWork3/main.py [hunk 22]
```diff
@@ -0,0 +1,239 @@
...
+                most_common_name = find_most_common_name(rearranged_data, decade, country)
+                with open(f'{max_age}_{avg_registered}_{most_common_name}.csv', 'w', newline='') as file:
+                    writer = csv.DictWriter(file, fieldnames=rearranged_data[decade][country][0].keys())
+                    writer.writeheader()
+                    for elem in rearranged_data[decade][country]:
+                        writer.writerow(elem)
+            except KeyError:
+                logger.warning(f'No country {country} in {decade}')
+    logger.info('Successfully wrote users data to local files')
+    os.chdir(curr_dir)
```
Comment (vloooo) [inline HomeWork3/main.py]: can be improved

File: HomeWork3/main.py (hunk 9 above)
Comment (Uaplexer) [reply HomeWork3/main.py]: something creating an error without that, maybe because the format is 0:00 and not 00:00

File: HomeWork3/main.py [hunk 23]
```diff
@@ -0,0 +1,249 @@
...
+        writer.writerows(data)
+
+
+def read_csv_file():
+    with open(EXACT_CSV_FILENAME, 'r', newline='') as csvfile:
+        return list(csv.DictReader(csvfile))
+
+
+def download_csv_file():
+    url = 'https://randomuser.me/api/?results=5000&format=csv'
```
Comment (vloooo) [inline HomeWork3/main.py]: make as a constant

File: HomeWork3/main.py [hunk 24]
```diff
@@ -0,0 +1,249 @@
...
+        logger.info('Data was filtered by gender')
+        return filtered_data
+    elif num_of_rows and num_of_rows <= len(csv_data):
+        filtered_data = csv_data[:num_of_rows]
+        logger.info('Data was filtered by number of rows')
+        return filtered_data
+    else:
+        filtered_data = csv_data
+        logger.warning("Data wasn't filtered by any of params")
+        return filtered_data
```
Comment (vloooo) [inline HomeWork3/main.py]: )))))
```suggestion
def filter_csv(csv_data, gender=None, num_of_rows=None):
    if gender:
        csv_data = list(filter(lambda x: x['gender'] == gender, csv_data))
        logger.info('Data was filtered by gender')
    elif num_of_rows and num_of_rows <= len(csv_data):
        csv_data = csv_data[:num_of_rows]
        logger.info('Data was filtered by number of rows')
    else:
        logger.warning("Data wasn't filtered by any of params")
    return csv_data
```

File: HomeWork3/main.py (hunk 9 above)
Comment (vloooo) [reply HomeWork3/main.py]: let's see

File: HomeWork3/main.py [hunk 25]
```diff
@@ -0,0 +1,249 @@
...
+        move(os.path.join(curr_path_folder, EXACT_CSV_FILENAME), f'{program_files_folder}Users_data_folder')
+        logger.info(f'Successfully moved file to default directory "{program_files_folder}Users_data_folder"')
+    else:
+        move(os.path.join(os.getcwd(), EXACT_CSV_FILENAME), path_to_dir)
+        logger.info('Successfully moved initial file to selected directory')
+
+        os.chdir(path_to_dir)
+        os.makedirs('Users_data_folder', exist_ok=True)
+        logger.info('Successfully created folder "Users_data_folder"')
+        os.chdir(f'{path_to_dir}\\Users_data_folder')
```
Comment (vloooo) [inline HomeWork3/main.py]: too complicated)
let's try to refactor together

File: HomeWork3/main.py [hunk 26]
```diff
@@ -0,0 +1,249 @@
...
+
+
+def rearrange_data(csv_data):
+    new_data = {}
+    for elem in csv_data:
+        decade = f'{elem["dob.date"][-2]}0-th'
+        new_data.setdefault(decade, {})
+        new_data[decade].setdefault(elem['location.country'], [])
+        new_data[decade][elem['location.country']].append(elem)
```
Comment (vloooo) [inline HomeWork3/main.py]: elem['location.country'] could be variable

File: HomeWork3/main.py [hunk 27]
```diff
@@ -0,0 +1,249 @@
...
+def get_countries(csv_data):
+    return set([elem['location.country'] for elem in csv_data])
+
+
+def create_sub_folders(decades, countries):
+    curr_dir = os.getcwd()
+
+    for decade in decades:
+        decade_dir = os.path.join(curr_dir, decade)
+        os.makedirs(decade_dir, exist_ok=True)
```
Comment (vloooo) [inline HomeWork3/main.py]: let's look together

File: HomeWork3/main.py [hunk 28]
```diff
@@ -0,0 +1,249 @@
...
+    logger.info('Successfully rearranged data!')
+    return new_data
+
+
+def get_decades(csv_data):
+    return csv_data.keys()
+
+
+def get_countries(csv_data):
+    return set([elem['location.country'] for elem in csv_data])
```
Comment (vloooo) [inline HomeWork3/main.py]: I don't think these methods are helpful.
I suspect get_countries even create a mistake

File: HomeWork3/main.py [hunk 29]
```diff
@@ -0,0 +1,249 @@
...
+        for country in countries:
+            country_dir = os.path.join(decade_dir, country)
+            os.makedirs(country_dir, exist_ok=True)
+
+    os.chdir(curr_dir)
+    logger.info('Successfully created sub folders with Countries and Decades')
+
+
+def find_max_age(rearranged_csv, decade, country):
+    return min([int(elem['dob.date'][6:]) for elem in rearranged_csv[decade][country]])
```
Comment (vloooo) [inline HomeWork3/main.py]: interesting)
let's look

File: HomeWork3/main.py [hunk 30]
```diff
@@ -0,0 +1,249 @@
...
+
+def write_users_data_to_different_files(rearranged_data, decades, countries):
+    curr_dir = os.getcwd()
+
+    for decade in decades:
+        decade_dir = os.path.join(curr_dir, decade)
+        logger.info(f'Successfully joined folder {decade}')
+        for country in countries:
+            country_dir = os.path.join(decade_dir, country)
+            logger.info(f'\t\tSuccessfully joined folder {country}')
```
Comment (vloooo) [inline HomeWork3/main.py]: you already had such loops let's try to reuse them

File: HomeWork3/main.py [hunk 31]
```diff
@@ -0,0 +1,249 @@
...
+                avg_registered = find_avg_registered(rearranged_data, decade, country)
+                most_common_name = find_most_common_name(rearranged_data, decade, country)
+
+                file_name = f'{max_age}_{avg_registered}_{most_common_name}.csv'
+
+                with open(file_name, 'w', newline='') as file:
+                    writer = csv.DictWriter(file, fieldnames=rearranged_data[decade][country][0].keys())
+                    writer.writeheader()
+                    writer.writerows(rearranged_data[decade][country])
+                logger.info(f'\t\t\tSuccessfully created file {file_name}')
```
Comment (vloooo) [inline HomeWork3/main.py]: these are two more functions

File: HomeWork3/main.py [hunk 32]
```diff
@@ -0,0 +1,249 @@
...
+
+            except KeyError:
+                logger.warning(f'No country {country} in {decade}')
+
+    logger.info('Successfully wrote users data to local files')
+    os.chdir(curr_dir)
```
Comment (vloooo) [inline HomeWork3/main.py]: could be avoided

File: HomeWork3/main.py [hunk 33]
```diff
@@ -0,0 +1,249 @@
...
+            rmtree(os.path.join(os.getcwd(), decade))
+            logger.info(f'Successfully deleted {decade} decade')
+
+
+# This func is creating infinite zip
+def archive_dest_folder(path_to_folder):
+    if os.path.exists(path_to_folder):
+        make_archive(f'{path_to_folder}\\Users_data_folder', 'zip')
+    else:
+        make_archive('C:\\Program Files\\Users_data_folder', 'zip')
```
Comment (vloooo) [inline HomeWork3/main.py]: )))))
let's see how it could be improved

File: HomeWork3/main.py [hunk 34]
```diff
@@ -0,0 +1,249 @@
...
+
+
+# Task 3
+args = parse_args()
+EXACT_CSV_FILENAME = args.filename + '.csv'
```
Comment (vloooo) [inline HomeWork3/main.py]: it's not a constant.
also use f-string

File: HomeWork3/main.py [hunk 35]
```diff
@@ -0,0 +1,214 @@
...
+from shutil import move, rmtree, make_archive
+from collections import Counter
+from statistics import mean
+
+URL = 'https://randomuser.me/api/?results=5000&format=csv'
+
+
+def parse_args():
+    parser = argp.ArgumentParser()
+    parser.add_argument('filename', type=str, default='output', help='Name of data file')
```
Comment (vloooo) [inline HomeWork3/main.py]: I've never said filename had to be positional)))

File: HomeWork3/main.py [hunk 36]
```diff
@@ -0,0 +1,214 @@
...
+
+
+def add_and_change_fields_in_file(csv_data):
+    for i, elem in enumerate(csv_data, start=1):
+        elem['global_index'] = i
+        elem['current_time'] = current_time_implementation(elem['location.timezone.offset'])
+        elem['name.title'] = name_title_implementation(elem['name.title'])
+        elem['dob.date'] = datetime.strptime(elem['dob.date'], '%Y-%m-%dT%H:%M:%S.%fZ').strftime('%m/%d/%Y')
+        elem['registered.date'] = datetime.strptime(elem['registered.date'], '%Y-%m-%dT%H:%M:%S.%fZ').strftime(
+            '%m-%d-%Y, %H:%M:%S')
```
Comment (vloooo) [inline HomeWork3/main.py]: looks like a function with 2 parameters

File: HomeWork3/main.py [hunk 37]
```diff
@@ -0,0 +1,214 @@
...
+        logger.info('Successfully created folder "Users_data_folder"')
+
+        move(os.path.join(curr_path_folder, filename), f'{curr_path_folder}\\Users_data_folder')
+        logger.info(f'Successfully moved file to default directory "{curr_path_folder}\\Users_data_folder"')
+    else:
+        os.makedirs(os.path.join(path_to_dir, 'Users_data_folder'), exist_ok=True)
+        logger.info('Successfully created folder "Users_data_folder"')
+
+        move(os.path.join(curr_path_folder, filename), path_to_dir)
+        logger.info('Successfully moved initial file to selected directory')
```
Comment (vloooo) [inline HomeWork3/main.py]: ))) let's discuss this part again)

File: HomeWork3/main.py [hunk 38]
```diff
@@ -0,0 +1,214 @@
...
+            '%m-%d-%Y, %H:%M:%S')
+    logger.info('Successfully added and modified attributes')
+    return csv_data
+
+
+def create_and_move_file_to_destination_folder(path_to_dir, filename):
+    curr_path_folder = os.getcwd()
+    if not os.path.exists(path_to_dir):
+        logger.warning('Wrong path was entered in args!')
+        os.makedirs("Users_data_folder", exist_ok=True)
```
Comment (vloooo) [inline HomeWork3/main.py]: BTW, double-quotes ;)

File: HomeWork3/main.py [hunk 39]
```diff
@@ -0,0 +1,214 @@
...
+        new_data.setdefault(decade, {})
+        new_data[decade].setdefault(country, [])
+        new_data[decade][country].append(elem)
+    logger.info('Successfully rearranged data!')
+    return new_data
+
+
+def get_decades(csv_data):
+    # ['00-th', '90-th', '80-th', '70-th', '60-th', '50-th', '40-th']
+    return csv_data.keys()
```
Comment (vloooo) [inline HomeWork3/main.py]: I thought we agreed to delete this

File: HomeWork3/main.py [hunk 40]
```diff
@@ -0,0 +1,214 @@
...
+    most_common_name = find_most_common_name(rearranged_csv, decade, country)
+    return max_age, avg_registered, most_common_name
+
+
+def create_file(file_path, file_name, data):
+    with open(file_path, 'w', newline='') as file:
+        writer = csv.DictWriter(file, fieldnames=data[0].keys())
+        writer.writeheader()
+        writer.writerows(data)
+    logger.info(f'\t\tSuccessfully created file {file_name}')
```
Comment (vloooo) [inline HomeWork3/main.py]: what's the difference between the function **write_to_csv**?

File: HomeWork3/main.py [hunk 41]
```diff
@@ -0,0 +1,214 @@
...
+    for decade in decades:
+        logger.info(f'Successfully joined folder {decade}')
+        for country in get_countries(rearranged_data, decade):
+            country_dir = os.path.join(curr_dir, decade, country)
+            os.makedirs(country_dir, exist_ok=True)
+            logger.info(f'\tSuccessfully joined folder {country}')
+            max_age, avg_registered, most_common_name = get_file_name(rearranged_data, decade, country)
+            file_name = f'{max_age}_{avg_registered:.0f}_{most_common_name}.csv'
+            file_path = os.path.join(country_dir, file_name)
+            create_file(file_path, file_name, rearranged_data[decade][country])
```
Comment (vloooo) [inline HomeWork3/main.py]: let's refactor this code together

File: HomeWork3/main.py [hunk 42]
```diff
@@ -0,0 +1,214 @@
...
+create_and_move_file_to_destination_folder(args.path, csv_filename)
+# Task 8
+rearranged = rearrange_data(modified_data)
+decades = get_decades(rearranged)
+# Task 9-11,13
+write_users_data_to_different_files(rearranged, decades)
+# Task 12
+delete_decades_under_60(decades)
+# Task 14
+archive_dest_folder(args.path)
```
Comment (vloooo) [inline HomeWork3/main.py]: image

**Review by vloooo — APPROVED** [review]
Great job!
wait for your next tasks

**Pull request merged**